WordPress/Classes/Services/PostService+RefreshStatus.swift

```diff
@@ -9,22 +9,24 @@ extension PostService {
     ///
     func refreshPostStatus(onCompletion: (() -> Void)? = nil, onError: ((Error) -> Void)? = nil) {
         self.managedObjectContext.perform {
-            let fetch = NSFetchRequest<Post>(entityName: Post.classNameWithoutNamespaces())
+            let request = NSBatchUpdateRequest(entityName: Post.classNameWithoutNamespaces())
             let pushingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushing.rawValue))
             let processingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushingMedia.rawValue))
-            fetch.predicate = NSCompoundPredicate(orPredicateWithSubpredicates: [pushingPredicate, processingPredicate])
+            let pushingOrProcessingPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [pushingPredicate, processingPredicate])
+            let notFailedPredicate = NSPredicate(format: "remoteStatusNumber != %@ AND NOT (postID != nil AND postID > 0)", NSNumber(value: AbstractPostRemoteStatus.failed.rawValue))
+            request.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [pushingOrProcessingPredicate, notFailedPredicate])
+            request.propertiesToUpdate = [
+                "remoteStatusNumber": NSNumber(value: AbstractPostRemoteStatus.failed.rawValue),
+                "status": NSString(string: BasePost.Status.draft.rawValue),
```

Comment:
Why are we updating to draft?

Reply:
in the `markAsFailedAndDraftIfNeeded` method, if the post does not have remote, its `status` is set to draft and `dateModified` to new date

Reply:
That happens only if the post is a page

Reply:
I now see that there is a different implementation of the `markAsFailedAndDraftIfNeeded` method. this is the implementation I have seen when I've submitted this commit:

<img width="489" alt="markAsFailedAndDraftIfNeeded" src="https://user-images.githubusercontent.com/54806784/68164990-867f2900-ff5e-11e9-8a3b-c653e8a0f5c1.png">

[According to Apple](https://developer.apple.com/library/archive/featuredarticles/CoreData_Batch_Guide/BatchUpdates/BatchUpdates.html) however, NSBatchUpdateRequest can not contain any logic beside the predicate, what this means I think is that the posts would have to be updated in two batches now -- one batch for the posts that are not failed (and set them to failed) and the other posts that do not have a remote and are a Page. would that be alright?

Reply:
hey @yaelirub, I've changed the current implementation so that it conforms to the `markAsFailedAndDraftIfNeeded()` method. it has to be done in two batches, which is probably still faster than the fetch. 

however, I've uncovered an unwanted behavior (at least I assume it's unwanted). the fetched entities were of type `Post: AbstractPost` and these entities were then checked whether they're of type `Page: AbstractPost`, which would never result in true, since pages would not be fetched at all. I've tested this. so I've changed the fetched entity to `AbstractPost`. please let me know if I should revert it back to `Post` in case I'm missing something.